pydantic/_internal/_fields.py

```diff
@@ -264,3 +267,15 @@ def collect_fields(  # noqa: C901
             field.annotation = replace_types(field.annotation, typevars_map)
 
     return fields, class_vars
+
+
+def _is_finalvar_with_default_val(type_: type[Any], val: Any) -> bool:
+    from pydantic.fields import FieldInfo
```

Comment:
don't like needing the local import but couldn't see anywhere it could go that would eliminate this, and putting this function in this file seemed like the best way to do it

Reply:
fine by me.